helix-core/src/main/java/org/apache/helix/HelixManagerProperty.java

```diff
@@ -0,0 +1,75 @@
+package org.apache.helix;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Properties;
+import org.apache.helix.model.CloudConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Hold Helix participant properties. The participant properties further hold Helix cloud properties and some other properties
+ * specific for the participant.
+ */
+public class HelixManagerProperty {
+  private static final Logger LOG = LoggerFactory.getLogger(HelixManagerProperty.class.getName());
+  private String _version;
+  private long _healthReportLatency;
+  private HelixCloudProperty _helixCloudProperty;
+
+  /**
+   * Initialize Helix manager property with default value
+   * @param properties helix manager related properties input as a map
+   * @param cloudConfig cloudConfig read from Zookeeper
+   */
+  public HelixManagerProperty(Properties properties, CloudConfig cloudConfig) {
+    if (cloudConfig != null) {
+      _helixCloudProperty = new HelixCloudProperty(cloudConfig);
```

Comment:
I think we discussed this once, if we have a boolean "isInCloud" (I think we had a better name, but I don't remember), then this nullable value can be handled more gracefully.
At least we can check if the "isInCloud" == true, null cloudconfig input should trigger an exception.

Reply:
The cloudEnabled flag is actually in cloud config itself. Remember that we talk about we do not want to touch cluster config, and we put all cloud related configs in a separate znode.

Reply:
As we discussed, my major concern is getHelixCloudProperty() will return null. This makes the class harder to use.

Reply:
Changed to have a default cloudconfig value.